pkg/validation/internal/operatorhub.go

```diff
@@ -81,41 +101,202 @@ func validateBundleOperatorHub(bundle *manifests.Bundle) errors.ManifestResult {
 	for _, warn := range warns {
 		result.Add(errors.WarnInvalidCSV(warn.Error(), bundle.CSV.GetName()))
 	}
+
+	errs, warns = validateHubDeprecatedAPIS(bundle, k8sVersion)
+	for _, err := range errs {
+		result.Add(errors.ErrFailedValidation(err.Error(), bundle.CSV.GetName()))
+	}
+	for _, warn := range warns {
+		result.Add(errors.WarnFailedValidation(warn.Error(), bundle.CSV.GetName()))
+	}
+
+	if warn := validateHubChannels(bundle.Channels); warn != nil {
+		result.Add(errors.WarnFailedValidation(warn.Error(), bundle.CSV.GetName()))
+	}
+
 	return result
 }
 
-func validateHubCSVSpec(csv v1alpha1.ClusterServiceVersion) ([]error, []error) {
+// validateHubChannels will check the channels. The motivation for the following check is to ensure that operators
+// authors knows if their operator bundles are or not respecting the Naming Convention Rules.
+// However, the operator authors still able to choose the names as please them.
+func validateHubChannels(channels []string) error {
+	const preview = "preview"
+	const stable = "stable"
+	const fast = "fast"
+	for _, channel := range channels {
+		if !strings.Contains(channel, preview) &&
+			!strings.Contains(channel, stable) &&
+			!strings.Contains(channel, fast) {
+			return fmt.Errorf("ensure that your channels are following the channel naming convention. " +
+				"More info: https://olm.operatorframework.io/docs/best-practices/channel-naming/")
+		}
+
+	}
+	return nil
+}
+
+// validateHubDeprecatedAPIS will check if the operator bundle is using a deprecated or no longer supported k8s api
+func validateHubDeprecatedAPIS(bundle *manifests.Bundle, k8sVersion string) ([]error, []error) {
 	var errs []error
 	var warns []error
+	k8s1_22, _ := semver.Make("1.22.0")
```

Comment:
Can we comment on what these semver versions mean? Why do these k8s versions matter in the grand scheme of things? I can guess these are upper and lower bounds for k8s version.
Perhaps we should convert those two to const so we update them in the future?

Reply:
done